test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -599,20 +602,61 @@ public int liveMemberCount()
     public NodeToolResult nodetoolResult(boolean withNotifications, String... commandAndArgs)
     {
         return sync(() -> {
-            DTestNodeTool nodetool = new DTestNodeTool(withNotifications);
-            int rc =  nodetool.execute(commandAndArgs);
-            return new NodeToolResult(commandAndArgs, rc, new ArrayList<>(nodetool.notifications.notifications), nodetool.latestError);
+            try (CapturingConsoleOutputProvider provider = new CapturingConsoleOutputProvider())
+            {
+                DTestNodeTool nodetool = new DTestNodeTool(withNotifications, provider);
+                int rc = nodetool.execute(commandAndArgs);
+                return new NodeToolResult(commandAndArgs, rc,
+                                          new ArrayList<>(nodetool.notifications.notifications),
+                                          nodetool.latestError,
+                                          provider.getOutString(),
+                                          provider.getErrString());
+            }
         }).call();
     }
 
+    private static class CapturingConsoleOutputProvider implements ConsoleOutputProvider
+    {
+        private final ByteArrayOutputStream outBase = new ByteArrayOutputStream();
+        private final ByteArrayOutputStream errBase = new ByteArrayOutputStream();
+        private final PrintStream out = new PrintStream(outBase, true);
+        private final PrintStream err = new PrintStream(errBase, true);
+
+        public PrintStream outStream()
+        {
+            return out;
+        }
+
+        public PrintStream errStream()
+        {
+            return err;
+        }
+
+        public String getOutString()
+        {
+            return outBase.toString();
+        }
+
+        public String getErrString()
+        {
+            return errBase.toString();
+        }
+
+        public void close()
```

Comment:
don't need to close `ByteArrayOutputStream`, here is the implementation of close http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/io/ByteArrayOutputStream.java#l267